php/commands/core.php

```diff
@@ -1150,6 +1169,30 @@ private function get_updates( $assoc_args ) {
 		return array_values( $updates );
 	}
 
+	private function remove_unused_files( $version_from, $version_to, $locale ) {
+		if ( ! $version_from || ! $version_to ) {
+			return;
+		}
+
+		$old_checksums = self::get_core_checksums( $version_from, $locale ? $locale : 'en_US' );
+		$new_checksums = self::get_core_checksums( $version_to, $locale ? $locale : 'en_US' );
+
+		if ( empty( $old_checksums ) || empty( $new_checksums ) ) {
+			return;
+		}
+
+		$files_to_remove = array_diff( array_keys( $old_checksums ), array_keys( $new_checksums ) );
+
+		if ( ! empty( $files_to_remove ) ) {
+			foreach ( $files_to_remove as $file ) {
+				if ( file_exists( ABSPATH . $file ) ) {
+					unlink( ABSPATH . $file );
+				}
+			}
+			WP_CLI::log( count($files_to_remove) . ' files cleaned up' );
```

Comment:
I think we should log each of the files removed, to give greater visibility to the end user of what's happening. When something goes wrong, the count doesn't help me debug at all.